### ISSUE TITLE
When I Receive Blocks breaking Scratch 3 / Blockly

### ISSUE THREAD

**Issue by griffpatch** [LLK/scratch-vm#813]
### Expected Behavior

When loading a project containing <When I Receive [test1]> Blocks the GUI doesn't die, and receiver blocks are visible in GUI.

Load this project, switch to costumes then back to scripts:
https://llk.github.io/scratch-gui/develop/#188325200

### Actual Behavior

* Load Project
* Notice the broadcast block is visible, but the receiver block directly underneath does not appear?
* (Sometimes Error appears immediately in console at this point)
* Switch to Costumes Tab
* Switch Back to Blocks and screen goes entire browser window goes white and error appears in console:

```
lib.min.js:116 Uncaught (in promise) Error: Variable "test1" is already in use and its type is "" which conflicts with the passed in type, "broadcast_msg".
    at Blockly.VariableMap.createVariable (lib.min.js:116)
    at Blockly.WorkspaceSvg.Blockly.Workspace.createVariable (lib.min.js:116)
    at Blockly.WorkspaceSvg.createVariable (lib.min.js:116)
    at Blockly.FieldVariable.initModel (lib.min.js:116)
    at Blockly.FieldVariable.init (lib.min.js:116)
    at Blockly.Input.init (lib.min.js:116)
    at Blockly.BlockSvg.initSvg (lib.min.js:116)
    at Object.Blockly.Xml.domToBlock (lib.min.js:116)
    at Object.Blockly.Xml.domToWorkspace (lib.min.js:116)
    at Object.Blockly.Xml.clearWorkspaceAndLoadFromXml (lib.min.js:116)
```

This is the project I'm loading - A fresh project with just two blocks in it:
![image](https://user-images.githubusercontent.com/6737342/33164848-80f71a32-d02c-11e7-85ca-b94421038f99.png)

And this is how it appears in Scratch 3 GUI
![image](https://user-images.githubusercontent.com/6737342/33164882-a115908c-d02c-11e7-8196-f045f4310040.png)

### Operating System and Browser

Windows PC using Chrome latest

**Comment by thisandagain** [comment]
Thanks @griffpatch. This appears to be a regression that was introduced recently. @kchadha can you investigate?

**Comment by kchadha** [comment]
We are currently missing the functionality to import projects from SB2 that have broadcast message blocks because the ability to translate sb2 broadcasts into typed variables has not been implemented yet. This should be fixed once I get SB2 imports working with broadcast message blocks.

**Comment by kchadha** [comment]
Fixed with #829.
